Admin/FragmentAdmin.php

```diff
@@ -0,0 +1,273 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\ArticleBundle\Admin;
+
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\AdminBundle\Route\RouteCollection;
+use Sonata\AdminBundle\Show\ShowMapper;
+use Sonata\ArticleBundle\FragmentService\FragmentServiceInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author  Hugo Briand <briand@ekino.com>
+ */
+final class FragmentAdmin extends AbstractAdmin
+{
+    /**
+     * @var FragmentServiceInterface[]
+     */
+    private $fragmentServices = array();
```

Comment:
> shouldn't we work again on the block logic too ?

Yes, maybe also (I don't know the block bundle that much). And also for [the admin class](https://github.com/sonata-project/SonataAdminBundle/issues/3947), that is a very big service. The problem is not easy to solve BTW, but it would be great to give it some thought so that we don't : 

- have an interface, and an an abstract class that has empty methods for every method in the interface
- use inheritance when there might be a solution using composition (doctrine1 did this by using the active record pattern, now look at doctrine2, isn't it cleaner?)

So I think you won't be able to solve this right now, but it would be great to give it a thought : what if we grouped things in a namespace instead of grouping them in a single class?